digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -0,0 +1,167 @@
+                int rows = params.get("rows", int.class, 1);
+                String jobId = startJob(request, params, query, rows);
+                state.set(JOB_ID, jobId);
+                throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
+            }
+
+            // Poll for query job status
+            try (TDOperator op = TDOperator.fromConfig(params)) {
+
+                assert existingJobId.isPresent();
+
+                TDJobOperator job = op.newJobOperator(existingJobId.get());
+                TDJob jobInfo = job.getJobInfo();
+                TDJob.Status status = jobInfo.getStatus();
+                logger.debug("poll job status: {}: {}", existingJobId.get(), jobInfo);
+
+                if (!status.isFinished()) {
+                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
+                }
+
+                logger.debug("fetching poll job result: {}", existingJobId.get());
+                boolean done = fetchJobResult(job, status);
+
+                if (!done) {
+                    state.remove(JOB_ID);
+                    throw TaskExecutionException.ofNextPolling(pollInterval, ConfigElement.copyOf(state));
+                }
+
+                return TaskResult.empty(request);
+            }
+        }
+    }
+
+    private String startJob(TaskRequest request, Config params, String originalQuery, int rows)
+    {
+        String engine = params.get("engine", String.class, "presto");
+        if (!engine.equals("presto") && !engine.equals("hive")) {
+            throw new ConfigException("Unknown 'engine:' option (available options are: hive and presto): " + engine);
+        }
+
+        String query = "select count(*) >= " + rows +
+                " from (select 1 from (" + originalQuery + ") raw limit " + rows + ") sub";
+
+        try (TDOperator op = TDOperator.fromConfig(params)) {
+
+            int priority = params.get("priority", int.class, 0);  // TODO this should accept string (VERY_LOW, LOW, NORMAL, HIGH VERY_HIGH)
+            int jobRetry = params.get("job_retry", int.class, 0);
+
+            TDJobRequest req = new TDJobRequestBuilder()
+                    .setType(engine)
+                    .setDatabase(op.getDatabase())
+                    .setQuery(query)
+                    .setRetryLimit(jobRetry)
+                    .setPriority(priority)
+                    .setScheduledTime(request.getSessionTime().getEpochSecond())
+                    .createTDJobRequest();
+
+            TDJobOperator job = op.submitNewJob(req);
+            logger.info("Started {} job id={}:\n{}", engine, job.getJobId(), query);
+
+            return job.getJobId();
+        }
+    }
+
+    private boolean fetchJobResult(TDJobOperator job, TDJob.Status status)
+    {
+        if (status != TDJob.Status.SUCCESS) {
+            throw new RuntimeException("Poll job failed: " + job.getJobId());
```

Comment:
I think this is an expected error. So, here should throw TaskExecutionException so that OperatorManager suppresses stacktrace by default. Like TdOperatorFactory.joinJob method does. It also shows execution log of the job.